src/details/ArborX_DetailsBoundingVolumeHierarchyImpl.hpp

```diff
@@ -463,15 +184,15 @@ BoundingVolumeHierarchyImpl<DeviceType>::queryDispatch(
 
   Kokkos::Profiling::pushRegion("ArborX:BVH:sort_queries");
 
-  Kokkos::View<size_t *, DeviceType> permute;
+  Kokkos::View<size_t *, ExecutionSpace> permute;
```

Comment:
Why is `ExecutionSpace` here and not `MemorySpace`? There are 4 or 5 instances like that below.

Reply:
Because it does not matter (since we do not initialize in the view constructor and explicitly pass the execution space to `deep_copy()`) and because `ExecutionSpace` is there as a template parameter while I would need to somehow deduce the memory space.

Reply:
You literally wrote `using MemorySpace = typename BVH::memory_space;` 20 lines above, defined DeivceType, which you then used 3 lines below.

I would strongly prefer that we follow a single way of writing views internally in ArborX (memory spaces, or execution spaces) so that we don't spare any brain power each time trying to figure out if they are the same in a particular case.

Reply:
> You literally wrote `using MemorySpace = typename BVH::memory_space;` 20 lines above, defined DeivceType, which you then used 3 lines below.

Fair enough but that will be subject to change.  `TreeTraversal` should and will be templated directly on the bvh type and we may drop the template argument altogether in `BatchedQueries`.

Reply:
> I would strongly prefer that we follow a single way of writing views internally in ArborX (memory spaces, or execution spaces) so that we don't spare any brain power each time trying to figure out if they are the same in a particular case.

There is no one size fits all unfortunately.

Reply:
> I would strongly prefer that we follow a single way of writing views internally in ArborX (memory spaces, or execution spaces) so that we don't spare any brain power each time trying to figure out if they are the same in a particular case.

Why does it matter? I don't see the problem.

Reply:
If somebody writes `Kokkos::View<T*, Kokkos::Cuda>` and `Kokkos::View<T*, Kokkos::CudaSpace>` I have to do some mental gymnastics to see if they are the same. Things like, is UVM on?

Reply:
All views created in the body of the dispatch functions are templated on memory space now